helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixManager.java

```diff
@@ -202,6 +205,11 @@ public ZKHelixManager(String clusterName, String instanceName, InstanceType inst
 
   public ZKHelixManager(String clusterName, String instanceName, InstanceType instanceType,
       String zkAddress, HelixManagerStateListener stateListener) {
+    this(clusterName, instanceName, instanceType, zkAddress,stateListener, null);
+  }
+
+  public ZKHelixManager(String clusterName, String instanceName, InstanceType instanceType,
+      String zkAddress, HelixManagerStateListener stateListener, HelixParticipantProperty helixParticipantProperty) {
```

Comment:
ZKHelixManager is not only used by Participants. It could be created in many different modes such as Controller, Participant, Controller Participant, and Spectator. So I do not think adding HelixParticipantProperty in the constructor is a good idea because you're adding specificity to a general class. Using a more generic type would be better.

Reply:
HelixManagerProperty would be better.

Reply:
And we might want to deprecate the older constructor.